packages/xdl/src/Api.ts

```diff
@@ -101,33 +101,42 @@ async function _callMethodAsync(
       options = { ...options, ...requestOptions };
     }
   }
-  let response = await axios.request(options);
-  if (!response) {
-    throw new Error('Unexpected error: Request failed.');
-  }
-  let responseBody = response.data;
-  var responseObj;
-  if (isString(responseBody)) {
-    try {
-      responseObj = JSON.parse(responseBody);
-    } catch (e) {
-      throw new XDLError(
-        'INVALID_JSON',
-        'Invalid JSON returned from API: ' + e + '. Response body: ' + responseBody
+  try {
+    let response = await axios.request(options);
+    if (!response) {
+      throw new Error('Unexpected error: Request failed.');
+    }
+    let responseBody = response.data;
+    var responseObj;
+    if (isString(responseBody)) {
+      try {
+        responseObj = JSON.parse(responseBody);
+      } catch (e) {
+        throw new XDLError(
+          'INVALID_JSON',
+          'Invalid JSON returned from API: ' + e + '. Response body: ' + responseBody
+        );
+      }
+    } else {
+      responseObj = responseBody;
+    }
+    if (responseObj.err) {
+      let err = new ApiError(
+        responseObj.code || 'API_ERROR',
+        'API Response Error: ' + responseObj.err
       );
+      err.serverError = responseObj.err;
+      throw err;
+    } else {
+      return returnEntireResponse ? response : responseObj;
     }
-  } else {
-    responseObj = responseBody;
-  }
-  if (responseObj.err) {
-    let err = new ApiError(
-      responseObj.code || 'API_ERROR',
-      'API Response Error: ' + responseObj.err
+  } catch (error) {
+    const { response } = error;
+    const { ...errorObject } = response;
+    throw new Error(
```

Comment:
I think we should throw an `ApiError`, because we handle them differently from `Error` in that we don't show a stack trace.